src/contract/CallParams.ts

```diff
@@ -0,0 +1,751 @@
+                        this.func.addParamType({ ty: ArgumentType.uint64, array: true });
+                    } else {
+                        this.func.addParamType({ ty: ArgumentType.int64, array: true });
+                    }
+                } else if (param instanceof BigNumber) {
+                    if (!(param as BigNumber).isInteger()) {
+                        throw new TypeError("Floating point BigNumber is unsupported");
+                    }
+                    if ((param as BigNumber).isPositive()) {
+                        this.func.addParamType({ ty: ArgumentType.uint64, array: true });
+                    } else {
+                        this.func.addParamType({ ty: ArgumentType.int64, array: true });
+                    }
+                } else if (param instanceof Uint8Array) {
+                    switch ((param as Uint8Array).length) {
+                        case 1:
+                            this.func.addParamType({ ty: ArgumentType.uint8, array: false });
+                            break;
+                        case 2:
+                            this.func.addParamType({ ty: ArgumentType.uint16, array: false });
+                            break;
+                        case 4:
+                            this.func.addParamType({ ty: ArgumentType.uint32, array: false });
+                            break;
+                        case 8:
+                            this.func.addParamType({ ty: ArgumentType.uint64, array: false });
+                            break;
+                        case 20:
+                            this.func.addParamType({ ty: ArgumentType.address, array: false });
+                            break;
+                        case 24:
+                            this.func.addParamType({ ty: ArgumentType.func, array: false });
+                            break;
+                        case 32:
+                            this.func.addParamType({ ty: ArgumentType.uint256, array: false });
+                            break;
+                        default:
+                            if ((param as Uint8Array).length < 32) {
+                                this.func.addParamType(
+                                    { ty: ArgumentType.bytesfix, array: false },
+                                    (param as Uint8Array).length
+                                );
+                            } else {
+                                this.func.addParamType({ ty: ArgumentType.bytes, array: false });
+                            }
+                    }
+                } else {
+                    throw new TypeError("Should not be possible to get here");
+                }
+        }
+    }
+
+    private conditionallyAddType(ty: SoldityType, length?: number): void {
+        if (this.func.argumentTypes.length === this.currentArgument) {
+            this.func.addParamType(ty, length);
+        }
+    }
+
+    public addUint8(param: number): this {
```

Comment:
All these integer methods are overkill; in the Java SDK it's just one method that has the user specify the byte-width of the integer.